src/hockeypuck/pghkp/storage.go

```diff
@@ -91,6 +91,157 @@ var drConstraintsSQL = []string{
 	`DROP INDEX subkeys_rfp;`,
 }
 
+var crTempTablesSQL = []string{
+	`CREATE TEMPORARY TABLE IF NOT EXISTS keys_copyin (
+rfingerprint TEXT,
+doc jsonb,
+ctime TIMESTAMP WITH TIME ZONE,
+mtime TIMESTAMP WITH TIME ZONE,
+md5 TEXT,
+keywords tsvector
+)
+`,
+	`CREATE TEMPORARY TABLE IF NOT EXISTS subkeys_copyin (
+rfingerprint TEXT,
+rsubfp TEXT
+)
+`,
+	`CREATE TEMPORARY TABLE IF NOT EXISTS keys_checked (
+rfingerprint TEXT NOT NULL PRIMARY KEY,
+doc jsonb NOT NULL,
+ctime TIMESTAMP WITH TIME ZONE NOT NULL,
+mtime TIMESTAMP WITH TIME ZONE NOT NULL,
+md5 TEXT NOT NULL UNIQUE,
+keywords tsvector
+)
+`,
+	`CREATE TEMPORARY TABLE IF NOT EXISTS subkeys_checked (
+rfingerprint TEXT NOT NULL,
+rsubfp TEXT NOT NULL PRIMARY KEY
+)
+`,
+}
+
+var drTempTablesSQL = []string{
+	`DROP TABLE IF EXISTS subkeys_copyin CASCADE
+`,
+	`DROP TABLE IF EXISTS keys_copyin CASCADE
+`,
+	`DROP TABLE IF EXISTS subkeys_checked CASCADE
+`,
+	`DROP TABLE IF EXISTS keys_checked CASCADE
+`,
+}
+
+var bulkTxSQL = []string{
```

Comment:
DDL above is declared in arrays because they're run once sequentially in a loop. For DML, I think a separate constant with a descriptive name for each statement would be more readable than referencing by array offset.

Reply:
The DML in these strings is run in two invocations of a loop. Could have been a single loop invocation, but for debug-ability I broke it to two. See `bulkInsertCheckKeys(..)` and `bulkInsertCheckSubkeys(..)` (lines 810 and 786 respectively). and `bulkInsertSingleTx(..)` (line 754). Nevertheless, I do not use an index into the slice of strings.

I'll change this string slice to separate constants.

Reply:
Do you want to do `bulkStatsSQL` also?

Reply:
Did `bulkStatsSQL` too.